QRgen.java

```diff
@@ -0,0 +1,109 @@
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import net.glxn.qrgen.QRCode;
+
+public class QRgen extends JPanel implements ActionListener {
+
+    private final JButton genButton;
+    private final JLabel tableLabel;
+    private JComboBox tableNumber;
+    private String barcodeText;
+    private JLabel qrImage;
+
+    public QRgen() {
+        //construct preComponents
+        Integer[] tableNumberItems = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+        //construct components
+        genButton = new JButton("Generate");
+        tableLabel = new JLabel("Table #");
+        tableNumber = new JComboBox(tableNumberItems);
+        qrImage = new JLabel("QR Code Here");
+
+        //adjust size and set layout
+        setPreferredSize(new Dimension(600, 800));
+        setLayout(null);
+
+        //add components
+        add(genButton);
+        add(tableLabel);
+        add(tableNumber);
+        add(qrImage);
+
+        //set component bounds (only needed by Absolute Positioning)
+        genButton.setBounds(435, 100, 100, 20);
+        tableLabel.setBounds(100, 100, 50, 25);
+        tableNumber.setBounds(255, 100, 100, 25);
+        qrImage.setBounds(175,300,250,250);
+
+        genButton.addActionListener(this);
+    }
+
+    public void actionPerformed(ActionEvent e) {
+        Integer tableNum = (Integer) tableNumber.getSelectedItem();
+        switch (tableNum) {

```

Comment:
could we change this to not be deterministic, this would limit us to only 10 tables Ideally we need a service that has a method that takes an input parameter so we can hook it up to an api. This looks like a stand alone application.

Ideally we could make a static method on a QR gen service so we can call it 

so 
class QRgen {
    static ImageIcon generateBarcode(string baseUrl,  int tableNumber) {
      // make the barcode here and return it
    }
}
i'm not sure how to convert an Image icon to a image format (png or something) but since the user will be interacting through the react front end we will have to convert the image to something web friendly

Reply:
I did see some things in my Googling about saving the result as a .png file, if we had a path to save it too. Like, a single file path that could be overwritten every time the service is called. Then we'd just need to automatically pull from that path to display. I'm guessing we can easily connect the base URL to the table number seamlessly to make the complete URL. I'm open to ideas I can work on tomorrow morning.

Reply:
This can be refactored to be dynamically set. I will take a look.

Reply:
I refactored it but I could not test it. Could you share the gradle.build file with the package on there or the jars that you used to test?

Reply:
Shared in a new comment.